lib/queuePopulator/QueuePopulator.js

```diff
@@ -407,21 +407,34 @@ class QueuePopulator {
         });
     }
 
+    _setupExtensionZookeeper(extension, cb) {
+        return extension.setupZookeeper(err => {
+            if (err) {
+                return cb(err);
+            }
+            if (extension.createZkPath) {
+                return extension.createZkPath(cb);
+            }
+            return cb();
+        });
+    }
+
     _setupExtensions(cb) {
         return async.each(this._extensions, (ext, next) => {
```

Comment:
Personally I prefer to name the callback `next` when the tasks are done one after another, and `done` for parallel execution or otherwise out-of-order. Here it is the opposite (`next` for `async.each` and `done` for `async.series`) which may be confusing to read. Maybe either swapping the names or renaming the callbacks to more specific names could help.